assets/app/features/upgrade/controllers/upgrade7-openstack-services.controller.spec.js

```diff
@@ -0,0 +1,263 @@
+/*global bard $controller $httpBackend should assert openStackFactory $q $rootScope */
+describe('openStack Services Controller', function() {
+    var controller,
+        passingOpenStackServices = {
+            'services': true,
+        },
+        failingOpenStackServices = {
+            'services': false,
+        },
+        passingOpenStackBackup = {
+            'backup': true,
+        },
+        failingOpenStackBackup = {
+            'backup': false,
+        },
+        failingErrors = {
+            error_message: 'Authentication failure'
+        },
+        passingOpenStackServicesResponse = {
+            data: passingOpenStackServices
+        },
+        failingOpenStackServicesResponse = {
+            data: failingOpenStackServices
+        },
+        passingOpenStackBackupResponse = {
+            data: passingOpenStackBackup
+        },
+        failingOpenStackBackupResponse = {
+            data: failingOpenStackBackup
+        },
+        failingResponse = {
+            data: {
+                errors: failingErrors
+            }
+        };
+
+    beforeEach(function() {
+        //Setup the module and dependencies to be used.
+        bard.appModule('crowbarApp');
+        bard.inject('$controller', '$rootScope', 'openStackFactory', '$q', '$httpBackend');
+
+        //Create the controller
+        controller = $controller('Upgrade7OpenStackServicesController');
+
+        //Mock requests that are expected to be made
+        $httpBackend.expectGET('app/features/upgrade/i18n/en.json').respond({});
+        $httpBackend.flush();
+
+    });
+
+    // Verify no unexpected http call has been made
+    bard.verifyNoOutstandingHttpRequests();
+
+    it('should exist', function() {
+        should.exist(controller);
+    });
+
+    describe('openStackServices Model', function () {
+        it('should be defined', function () {
+            should.exist(controller.openStackServices);
+        });
+
+        it('is not completed by default', function() {
+            assert.isFalse(controller.openStackServices.completed);
+        });
+
+        it('is not valid by default', function() {
+            assert.isFalse(controller.openStackServices.valid);
+        });
+
+        it('is not running by default', function() {
+            assert.isFalse(controller.openStackServices.running);
+        });
+
```

Comment:
It should be tested because it's an attribute of the current controller. The fact that is shared as a parameter to suse-lazy-spinner directive is a separate story